libunbound/libworker.c

```diff
@@ -168,14 +168,11 @@ libworker_setup(struct ub_ctx* ctx, int is_bg, struct ub_event_base* eb)
 		hints_delete(w->env->hints);
 		w->env->hints = NULL;
 	}
-	if(cfg->ssl_upstream || (cfg->tls_cert_bundle && cfg->tls_cert_bundle[0]) || cfg->tls_win_cert) {
-		w->sslctx = connect_sslctx_create(NULL, NULL,
-			cfg->tls_cert_bundle, cfg->tls_win_cert);
-		if(!w->sslctx) {
-			/* to make the setup fail after unlock */
-			hints_delete(w->env->hints);
-			w->env->hints = NULL;
-		}
+	w->sslctx = connect_sslctx_create(NULL, NULL,
+		cfg->tls_cert_bundle, cfg->tls_win_cert);
+	if(!w->sslctx) {
+		libworker_delete(w);
+		return NULL;
```

Comment:
This return failure does not unlock the lock cfglock that happens on line L181. The solution that used to be used is suggested in the code block.
```suggestion
		/* to make the setup fail after unlock */
￼		hints_delete(w->env->hints);
￼		w->env->hints = NULL;
```